### PR TITLE
added vpc endpoint to access secrets manager

### PR DESCRIPTION
**Pull request by AgusPk**
# What's this PR do?

Solves #10 

### Summary by CodeRabbit

- New Feature: Added a new attribute `region` to the VPC module to specify the AWS region.
- New Feature: Introduced an AWS VPC endpoint for Secrets Manager, enhancing security by enabling private connections between the VPC and Secrets Manager. This endpoint is associated with the private subnets of the VPC and a specified security group.
- Refactor: Added a new variable "region" in the VPC module for better configurability and code reusability.

### CODE REVIEW

File: modules/vpc/variables.tf [hunk 1]
```diff
@@ -33,3 +33,9 @@ variable "tags" {
   type        = map(any)
   default     = {}
 }
+
+variable "region" {
```
Comment (ulises-jeremias) [inline modules/vpc/variables.tf]: modules should not receive the region 👌🏻 you can get is using 

```hcl
data "aws_region" "current" {}

${data.aws_region.current.name}
```

Comment (AgusPk) [reply modules/vpc/variables.tf]: nice! didn't know I could do that

**Review by ulises-jeremias — APPROVED** [review]
🔥

**Comment by ulises-jeremias** [comment]
@AgusPk just need to run terraform fmt in the file that is failing on CI, the rest is OK 👌🏻

**Pull request merged**